js/bundle_test.go

```diff
@@ -675,3 +696,85 @@ func TestBundleEnv(t *testing.T) {
+	t.Run("ok", func(t *testing.T) {
+		testCases := []struct {
+			cm      compiler.CompatibilityMode
+			script  string
+			exclaim string
+		}{
+			{compiler.CompatibilityModeExtended,
+				`import exclaim from "./exclaim.js";
+				export let options = { vus: 12345 };
+				export let file = open("./file.txt");
+				export default function() { return exclaim(file); };`,
+				`export default function(s) { return s + "!" };`},
+			{compiler.CompatibilityModeBase,
+				`var exclaim = require("./exclaim.js");
+				module.exports.options = { vus: 12345 };
+				module.exports.file = open("./file.txt");
+				module.exports.default = function() { return exclaim(module.exports.file); };`,
+				`module.exports.default = function(s) { return s + "!" };`},
+		}
+
+		for _, tc := range testCases {
+			tc := tc
+			t.Run(tc.cm.String(), func(t *testing.T) {
+				fs := afero.NewMemMapFs()
+				_ = fs.MkdirAll("/path/to", 0755)
+				_ = afero.WriteFile(fs, "/path/to/file.txt", []byte(`hi`), 0644)
+				_ = afero.WriteFile(fs, "/path/to/exclaim.js", []byte(tc.exclaim), 0644)
+
+				rtOpts := lib.RuntimeOptions{CompatibilityMode: null.StringFrom(tc.cm.String())}
+				b, err := getSimpleBundle("/path/to/script.js", tc.script, fs, rtOpts)
+				assert.NoError(t, err)
+
+				arc := b.makeArchive()
+
+				assert.Equal(t, "js", arc.Type)
+				assert.Equal(t, lib.Options{VUs: null.IntFrom(12345)}, arc.Options)
+				assert.Equal(t, "file:///path/to/script.js", arc.FilenameURL.String())
+				assert.Equal(t, tc.script, string(arc.Data))
+				assert.Equal(t, "file:///path/to/", arc.PwdURL.String())
+
+				exclaimData, err := afero.ReadFile(arc.Filesystems["file"], "/path/to/exclaim.js")
+				assert.NoError(t, err)
+				assert.Equal(t, tc.exclaim, string(exclaimData))
+
+				fileData, err := afero.ReadFile(arc.Filesystems["file"], "/path/to/file.txt")
+				assert.NoError(t, err)
+				assert.Equal(t, `hi`, string(fileData))
+				assert.Equal(t, consts.Version, arc.K6Version)
+				assert.Equal(t, tc.cm.String(), arc.CompatibilityMode)
+			})
+		}
+	})
+	t.Run("err", func(t *testing.T) {
+		testCases := []struct {
+			name       string
+			compatMode string
+			code       string
+			expErr     string
+		}{
+			{"InvalidCompat", "es1", `export default function() {};`,
+				`invalid compatibility mode "es1". Use: "extended", "base"`},
+			// ES6 modules are not supported
+			{"Modules", "base", `export default function() {};`,
+				"file://script.js: Line 1:1 Unexpected reserved word"},
+			// Arrow functions are not supported
+			{"ArrowFuncs", "base",
+				`module.exports.default = function() {}; () => {};`,
+				"file://script.js: Line 1:42 Unexpected token ) (and 1 more errors)"},
+		}
+
+		for _, tc := range testCases {
+			tc := tc
+			t.Run(tc.name, func(t *testing.T) {
+				rtOpts := lib.RuntimeOptions{CompatibilityMode: null.StringFrom(tc.compatMode)}
+				_, err := getArchive(tc.code, rtOpts)
+				assert.EqualError(t, err, tc.expErr)
+			})
+		}
+	})
```

Comment:
Isn't this https://github.com/loadimpact/k6/pull/1206/files#diff-f89c2c8c3a8f3e930d70e082ef776f34R120-R145

Reply:
Sorry, GitHub is just showing me the diff page with that link. Can you explain what you mean?

Reply:
It looks like you have the same code twice testing the same thing and the diff I linked is the other place where it exists, but apparently it doesn't scroll down automatically, you can do so manually and you will see the code that is highlighted around 1/5 of the way down

Looking at it again ... it seems like the difference is whether you are making an archive(as here) or not(as in the linked code), in which case I would argue to get it out in a separate test and have a subTestCases for calling `getArchive` and `getSimpleBundle`

Reply:
Hhmm true, these tests are almost equal.

I initially split `Bundle.makeArchive` into a standalone test in 471d3aad (this is now squashed into 7b7af07d), I think so that I could reuse a simpler `getArchive` helper for `TestNewBundleFromArchive`, but somehow ended up with the same test, which probably means both should be tested together.

Will fix.

Reply:
OK, I opted for removing these particular error tests (for `TestBundleMakeArchive`), since they were essentially duplicates of the `TestNewBundle` ones (the error returned is from `NewBundle`, not from `makeArchive`, so these tests were pointless).

Let me know how it looks after 7c611ec8.